### DISCUSSION TITLE
UTF-8 error using addFileFromStream function on vapor

### DISCUSSION THREAD

**Discussion by danielbalado** [maennchen/ZipStream-PHP#186] [Q&A]
Hello,
I have been using ZipStream-PHP (2.1.0) to generate a zip file from resources from as S3 based bucket (in our case, digital ocean). Everything worked perfectly while working localy, but I started experiencing some issues after deploying the feature to staging using laravel vapor. In the logs I see the following:

```
Fatal error: Uncaught Exception: Error encoding runtime JSON response: Malformed UTF-8 characters, possibly incorrectly encoded in /var/task/vendor/laravel/vapor-core/src/Runtime/NotifiesLambda.php:49
```

It seems this Exception is only triggered when using the function 'addFileFromStream'. Other methods to add files to the zip work fine. Here is my code for reference:

```php
$options = new Archive();
$options->setContentType('application/octet-stream; charset=utf-8');
$options->setContentDisposition('attachment; filename="package.zip'.'"');
$options->setZeroHeader(true);
$options->setSendHttpHeaders(true);

$zip = new ZipStream('package.zip', $options);

$source_disk = 'digital_ocean';
$file_path = 'pressroom/assets/15/8/lalique-press-file-soleil-lalique-enpdf-1620312406.pdf';

$file = Storage::disk($source_disk)->readStream($file_path);

$zip->addFileFromStream($file_path, $file);

$zip->finish();
```
I thought it could be related with the readStream function, so I tried loading the resource using fopen, but the result was the same (works locally but not on vapor).

Is there any action I can take to correct this encoding issue? I have also contacted the Vapor team to confirm if the issue would be in the readStream function, so I'll keep you posted if that's the case

**Comment by NicolasCARPi** [comment]
Can you try to modify [this line](https://github.com/laravel/vapor-core/blob/2.0/src/Runtime/NotifiesLambda.php#L46) and add JSON_INVALID_UTF8_IGNORE or JSON_INVALID_UTF8_SUBSTITUTE as json_encode options?

**Reply by danielbalado** [reply]
Thank you for your reply. Modifying the line with any of those options the file gets created, but the result zip file cannot be opened (The file has an unknown format or is damaged). I have the impression that the lambda function is truncating the data. The Vapor team replied that probably the payload or response is exceeding the allowed size by AWS, so I will probably have to look for another solution. I will keep testing anyway. I have also noticed that for larger files, the function tries to load all the data before showing the download dialog (It doesn't happen that way in the local environment, where the dialog is shown inmediately) causing a request timeout if the file is too big

**Reply by NicolasCARPi** [reply]
>  but the result zip file cannot be opened 

with which unzipper? try several.

**Reply by danielbalado** [reply]
I tried winrar and 7-zip. I'm attaching the test file.
[test.zip](https://github.com/maennchen/ZipStream-PHP/files/6656179/test.zip)

**Reply by NicolasCARPi** [reply]
can confirm file is garbage.
